### PR TITLE
Added CMake build system and documentation.

### PR DESCRIPTION
**Pull request by markjolah**
This is a CMake enabled fork of NCEP BUFRLIB.  
 * This should make installation in jedi-stack trivial.  Build instructions are in README.md.
 * The build provides a CMake package config file, making usage in CMake projects trivial.
 * This build system can serve as a template to add CMake build capabilities to stand-alone Fortran/C/CXX projects using modern CMake packaging techniques.  I tried to adhere to best practices.
 * We still may require adding a FindBufrlib.cmake file in ecbuild or elsewhere if we need to find an installed `libbufr.a` on a system that has installed directly from the original source which will obviously not come with a CMake package config file.
 * I made this repo public, as it is a fork of a public domain repo with build-system only modifications.

### CODE REVIEW

**Comment by mmiesch** [comment]
Thanks @markjolah and @srherbener with your work on this.  I have not been through the code yet but I have a question first.  Currently there is a build script in jedi-stack called build_nceplibs.sh and a subdirectory with source code (and more build scripts) called nceplibs.  This needs to be cleaned up and I wonder if it could be incorporated into this repo.  Perhaps this would help to address @srherbener thoughts on whether a separate repo is necessary - if it were used to build all the ncep libraries maybe a separate repo would be more justified?  Or maybe, this could all be moved to subdirectories in jedi-stack?

I do not really know what is in the jedi-stack nceplibs directory or who uses it - I just copied this and the build commands from @xinzhang8noaa 's previous docker repo.  Xin - is there a GitHub or other web site where this code is available?

**Comment by mmiesch** [comment]
I would also like to bring this repo to the attention of @david-rundle - David, can this be useful for you in developing the RTTOV jcsda repo - noting the description by @markjolah?
**This build system can serve as a template to add CMake build capabilities to stand-alone Fortran/C/CXX projects using modern CMake packaging techniques. I tried to adhere to best practices.**

**Comment by srherbener** [comment]
Mark O and I just discussed the concern I had with creating this new repo and I had been assuming that we need to keep Fortran and Python on the same version of the bufrlib library. However, that appears to be a false assumption. I think that as long as all Fortran programs link with the bufrlib version that is in this repository, and we always build the python interface in the py-ncepbufr repo using the bufrlib code in the py-ncepbufr then it doesn't matter if this repo and the py-ncepbufr repo have different versions of bufrlib. Do others agree with this?

When this repo is ready, we will also need to update the jedi-stack build scripts so that instead of copying libbufr.a from the py-ncepbufr build to do the install, we clone this repo and build/install from the clone.

If everyone finds this agreeable, then let's put the need for the jedi-stack change into a ZenHub issue.

**Comment by markjolah** [comment]
Thanks Steve for the good review.  I fixed all the problems you identified plus some others that became apparent.  Note the library target names will continue to have the _static and _shared postfix but only within the CMake code (each target must have a distinct name) but they will create actual libraries with just the "libbufr" name and the .a or .so extension as appropriate.

Note to test, I recommend using something like `_install` as the install prefix, CMake will treat this as relative to the source directory and make a new `_install` subdirectory.  That way testing doesn't pollute your `/usr/local` directory, and you can just delete the _install directory to test building and installing with different options enabled.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * Automatic install script
+```
+./tools/build.sh <install-prefix> <optional-cmake-args>
+```
+
+### CMake options
+
+The following CMake variables control the build:
+ * `BUILD_STATIC_LIBS` - Build static libraries. [default=ON]
+ * `BUILD_SHARED_LIBS` - Build static libraries. [default=OFF]
```
Comment (mmiesch) [inline README.md]: typo - second line should be build shared libraries

**Pull request merged**